packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -542,23 +542,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (!options.secret || options.secret === this.secret) {
+      return this.addSingleUserRotationForMasterSecret(options);
+    }
+
+    const id = 'RotationSingleUser';
+    return this.addSingleUserRotationForSecret(options.secret as unknown as Construct, id, options.secret, options);
+  }
+
+  private addSingleUserRotationForMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
     if (!this.secret) {
-      throw new Error('Cannot add single user rotation for a cluster without secret.');
+      throw new Error('Cannot add single user rotation for a cluster without master secret.');
     }
 
     const id = 'RotationSingleUser';
-    const existing = this.node.tryFindChild(id);
-    if (existing) {
-      throw new Error('A single user rotation was already added to this cluster.');
+    return this.addSingleUserRotationForSecret(this, id, this.secret, options);
+  }
+
+  private addSingleUserRotationForSecret(scope: Construct, id: string, secret: secretsmanager.ISecret, options: RotationSingleUserOptions)
+    : secretsmanager.SecretRotation {
+    // TODO: this is a hack! what's the right way to do it?
+    if ((scope as any).node.tryFindChild(id)) {
+      const secretId = this === scope ? 'master' : 'additional'; // Is there a better way to do this?
+      throw new Error(`A single user rotation for ${secretId} secret is already added to this cluster.`);
```

Comment:
Let's keep the "was" here:

```suggestion
      throw new Error(`A single user rotation for ${secretId} secret was already added to this cluster.`);
```

Reply:
done.